src/stepFunctions/commands/createStateMachineFromTemplate.ts

```diff
@@ -5,130 +5,61 @@
+    StateMachineTemplateQuickPickItem,
+    TemplateFormats,
+} from '../wizards/createStateMachineWizard'
 
 export async function createStateMachineFromTemplate(context: vscode.ExtensionContext) {
     const logger: Logger = getLogger()
 
-    const quickPick = picker.createQuickPick<StateMachineTemplateQuickPickItem>({
-        options: {
-            ignoreFocusOut: true,
-            title: localize('AWS.message.prompt.selectStateMachineTemplate.placeholder', 'Select a starter template'),
-            step: 1,
-            totalSteps: 1,
-        },
-        buttons: [vscode.QuickInputButtons.Back],
-        items: STARTER_TEMPLATES,
-    })
-
-    const choices = await picker.promptUser({
-        picker: quickPick,
-        onDidTriggerButton: (_, resolve) => {
-            resolve(undefined)
-        },
-    })
-
-    const selection = picker.verifySinglePickerOutput(choices)
-
-    // User pressed escape
-    if (selection === undefined) {
-        return
-    }
+    const wizardResponse = await new CreateStateMachineWizard().run()
 
-    try {
-        logger.debug(`User selected the ${selection.label} template.`)
+    if (wizardResponse && wizardResponse.template && wizardResponse.templateFormat) {
+        try {
+            logger.debug(
+                `User selected the ${wizardResponse.template.label} template of ${wizardResponse.templateFormat} format`
+            )
 
-        const textDocumentFromSelection = await getTextDocumentForSelectedItem(selection, context.extensionPath)
+            const textDocumentFromSelection = await getTextDocumentForSelectedItem(
+                wizardResponse.template,
+                context.extensionPath,
+                wizardResponse.templateFormat
+            )
 
-        vscode.window.showTextDocument(textDocumentFromSelection)
-    } catch (err) {
-        logger.error(err as Error)
-        vscode.window.showErrorMessage(
-            localize(
-                'AWS.message.error.stepfunctions.getTextDocumentForSelectedItem',
-                'There was an error creating the State Machine Template, check log for details.'
+            vscode.window.showTextDocument(textDocumentFromSelection)
+        } catch (err) {
+            logger.error(err as Error)
+            vscode.window.showErrorMessage(
+                localize(
+                    'AWS.message.error.stepfunctions.getTextDocumentForSelectedItem',
+                    'There was an error creating the State Machine Template, check log for details.'
+                )
             )
-        )
+        }
     }
 }
 
 async function getTextDocumentForSelectedItem(
     item: StateMachineTemplateQuickPickItem,
-    extensionPath: string
+    extensionPath: string,
+    format: string
 ): Promise<vscode.TextDocument> {
+    let content = await readFileAsString(path.join(extensionPath, 'templates', item.fileName))
+
+    if (format === TemplateFormats.YAML) {
+        content = safeDump(safeLoad(content))
```

Comment:
what's the purpose of this? A brief 1-line comment explaining the purpose may be justified.

Reply:
it converts json string to yaml string. Ok, I will add a line.